ocaml/xapi/xapi_host.ml

```diff
@@ -1171,6 +1171,24 @@ let install_server_certificate ~__context ~host ~certificate ~private_key ~certi
   (* sever all connections done with stunnel *)
   Xapi_mgmt_iface.reconfigure_stunnel ~__context
 
+let emergency_server_certificate_reset ~__context =
+  let generate_ssl_cert = "/opt/xensource/libexec/generate_ssl_cert" in
+
+  let args = [!Xapi_globs.xapissl_path; Helper_hostname.get_hostname ()] in
+  ignore @@ Forkhelpers.execute_command_get_output generate_ssl_cert args;
+
+  let self = Helpers.get_localhost ~__context in
+
+  (* Delete records of the server certificate in this host *)
+  Db.Certificate.get_all_records ~__context
```

Comment:
We should assume the worst-case situation and the main reason for having this command, which is to fix things up when your on a pool slave, the current certificate is bad, and because of that the master connection is broken. That means no DB calls are possible until this is fixed. So perhaps we need to reset stunnel first?

Reply:
I would like to avoid the case where the certificate metadata does not match the certificate that's actually installed. I would rather try to do the database calls and simply ignore the errors than not doing the calls at all.

Reply:
I forgot to reset stunnel :scream: